### DISCUSSION TITLE
What is OSS and free, and what is not?

### DISCUSSION THREAD

**Discussion by divoxx** [ariga/atlas#3246] [Q&A]
I'm currently evaluating using ent + atlas, and it's not very clear what exactly is open source and/or free and what is considered enterprise and/or paid.

For example, I recently hit a bug when migrating down (reverting a migration) and noticed that in the stacktrace there were files that were not part of the public atlas repository.

I totally understand and support companies having enterprise plans with additional, more complex, features behind a license, but having parts of the CLI / library be closed source is something I wasn't expecting, and might be the reason we choose something else. Not being able to debug an issue, fork the repo, fix it, contribute upstream, is discouraging, because it means I would need to depend from upstream prioritization and fixes for a tool that would be such an integral part of our dev and production workflow.

 Is that actually the case?

**Comment by a8m** [comment]
Hey @divoxx 👋 

Please go over these two docs and let me know if you have any more questions:
https://atlasgo.io/features
https://atlasgo.io/community-edition

**Reply by divoxx** [reply]
Thank you, while that does not really explain what is open source, only what requires a paid license, those pages are helpful. Thank you for sharing, but it does look like Atlas is not going to work for us at this moment.

**Reply by a8m** [reply]
The community edition is the OSS one - the code in this repository.

**Reply by divoxx** [reply]
The distributed binary had code included in it, files visible in a panic stacktrace, that are not open source and not available as part of this repository.

For example, this file doesn't exist in this repository:

```
/home/runner/work/atlas-enterprise/atlas-enterprise/cmd/atlas/internal/cmdapi/migrate_ent.go:1297 +0x3c6
```

Nonetheless, the lack of basic functionality such as `migrate down` in the community edition makes it unsuitable for our use. Also, enterprise features being closed source would also not work for us, as it would be impossible for us to fix any bugs we may hit in such a workflow critical piece of software.

If you're interested in feedback, I think providing all the basic features required for production use (such as `migrate down`), as part of the community edition, would make it a lot more attractive for people to try, which would probably lead into more people willing to pay for a license for the advanced features down the road.

Also, everything should be open source, even if it requires a license, as any large organization who works mainly with open source software does so because, among other things, it values the ability to audit any code running with access to their production environment, and be able to address issues in a timely manner without depending on the upstream.

**Reply by a8m** [reply]
> The distributed binary had code included in it, files visible in a panic stacktrace, that are not open source and not available as part of this repository.

That's right. The non-community binaries that are distributed contain non-OSS code. We maintain and publish many parts of our work to OSS (by the way, Ent is one of them), but some parts are not OSS, and that's what keeps the company alive. That's not going to change anytime soon.

> Also, enterprise features being closed source would also not work for us, as it would be impossible for us to fix any bugs we may hit in such a workflow critical piece of software.

That might not work for you, and that's totally fine. You are welcome to use the community version of Atlas which is licensed under Apache 2. However, if you find a bug, feel free to report it in the issue tracker, and one of us will look into it. If you need a higher SLA and more commitment from our side, we offer this type of support as well.

**Reply by divoxx** [reply]
I didn't know there was a separate distribution for community. Having a community-edition/open source only binary definitely helps alleviate some of the concerns, as far as being closed source.

Anyhow, I know no one is entitled to anything, it's your project and company, after all. I just wanted to provide a perspective of a new user and potential large customer, and provide feedback on why it will unfortunately not work for a organization. It's really up to you if you find the feedback valuable.

To wrap it up, in case it's something you find valuable, I think the lack of `migrate down` in the community edition is a big drawback. That is a basic feature that anyone will want/need, and putting that behind a paywall makes it less likely that people will spend their time migrating their project and using the community edition, and then potentially converting to a paid user for all the other awesome features you provide.

Thanks for the help and clarifications!

**Reply by a8m** [reply]
Thanks for the feedback.
